### PR TITLE
Run example

### PR DESCRIPTION
**Pull request by byroot**
Includes #20 plus a bit of refactoring, mostly for ease of use in specs and error output readability:

capture d ecran 2015-08-27 a 12 31 46

It assert both STDOUT and STDERR output. So we expect warnings too.

@grosser thoughts?

### CODE REVIEW

File: spec/support/expect_script_helper.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  def failure_message
+    "expected:\n#{indented_source}\nto #{description}, but #{positive_failure_reason}"
+  end
+
+  def failure_message_when_negated
+    "expected:\n#{indented_source}\nto not #{description}, but #{negative_failure_reason}"
+  end
+
+  def indented_source
+    indent = @source.split("\n").select {|line| !line.strip.empty? }.map {|line| line.index(/[^\s]/) }.compact.min || 0
```
Comment (grosser) [inline spec/support/expect_script_helper.rb:19]: same as this `@source.gsub(/^ {#{@source[^ +].size}}/)` ?

File: spec/support/expect_script_helper.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    @source.gsub(/^[[:blank:]]{#{indent}}/, '  ')
+  end
+
+  def in_tmpdir(&block)
+    Dir.mktmpdir('bootscale') { |dir| Dir.chdir(dir, &block) }
+  end
+
+  def run_example(code)
+    in_tmpdir do
+      files = code.split(/^ +# (.*\.rb)\n/)[1..-1].each_slice(2).to_a
```
Comment (grosser) [inline spec/support/expect_script_helper.rb:29]: not sure how this is working when the indentation is supposed to be removed since it matches on `^ +`

**Comment by grosser** [comment]
I find the `expect(run_code).to eq "xxx"` easier to read / understand then this echo matcher ...

File: spec/support/expect_script_helper.rb (hunk 2 above)
Comment (byroot) [inline spec/support/expect_script_helper.rb:29]: Indentation is removed later and only to display the error message.

**Comment by grosser** [comment]
don't really care either way ... just want to use the include code thingy ... so if you like this more :+1:

File: spec/support/expect_script_helper.rb (hunk 2 above)
Comment (grosser) [inline spec/support/expect_script_helper.rb:29]: If it's removed upfront everything operates/prints on the same code ?

**Comment by byroot** [comment]
I understand your point. But I think having the source in the error message helps a lot. But yeah it's bikeshedding at this point.

**Pull request merged**